### ISSUE TITLE
Please provide the experimental environment and related packages for reproduction.

### ISSUE THREAD

**Issue by shygejiayou** [bt-nghia/DisCo#1]
很棒的工作！

请问能否提供一下实验环境和依赖的packages版本，以便我能复现结果并改进？

**Comment by bt-nghia** [comment]
Hi sorry for the late reply, are you able to reproduce the result? I run all my experiments on Colab or Kaggle. All configs are in config.py file.

**Comment by bt-nghia** [comment]
Oh Sorry I realise that I wrote this code in JAX-FLAX, did you have any trouble with that? here is my env settings for this:
----------------------------- -----------
absl-py                       2.1.0
accelerate                    1.10.0
aiohttp                       3.9.3
aiosignal                     1.3.1
alabaster                     1.0.0
appdirs                       1.4.4
archspec                      0.2.3
asttokens                     2.4.1
astunparse                    1.6.3
attrs                         23.2.0
babel                         2.16.0
boltons                       23.0.0
box2d-py                      2.3.5
Brotli                        1.0.9
certifi                       2024.2.2
cffi                          1.16.0
charset-normalizer            2.0.4
chex                          0.1.86
click                         8.1.7
clikit                        0.6.2
cloudpickle                   3.0.0
comm                          0.2.2
conda                         24.3.0
conda-content-trust           0.2.0
conda-libmamba-solver         23.12.0
conda-package-handling        2.2.0
conda_package_streaming       0.9.0
contourpy                     1.2.1
cornac                        2.2.2
crashtest                     0.3.1
cryptography                  41.0.7
cycler                        0.12.1
Cython                        0.29.37
datasets                      3.6.0
debugpy                       1.6.7
decorator                     5.1.1
deepctr-torch                 0.2.9
diffusers                     0.32.1
dill                          0.3.8
distrax                       0.1.5
distro                        1.8.0
dm-env                        1.6
dm-tree                       0.1.8
docker-pycreds                0.4.0
docutils                      0.21.2
et_xmlfile                    2.0.0
etils                         1.8.0
exceptiongroup                1.2.0
executing                     2.0.1
Farama-Notifications          0.0.4
fasteners                     0.19
filelock                      3.13.1
flatbuffers                   25.2.10
flax                          0.8.2
fonttools                     4.51.0
frozenlist                    1.4.1
fsspec                        2024.2.0
gast                          0.6.0
gcc7                          0.0.7
gitdb                         4.0.11
GitPython                     3.1.43
glfw                          2.7.0
google-pasta                  0.2.0
grpcio                        1.62.1
gym                           0.9.1
gym-notices                   0.0.8
gymnasium                     0.29.1
h5py                          3.12.1
hf-xet                        1.1.2
httpstan                      4.13.0
huggingface-hub               0.32.0
idna                          3.4
ImageHash                     4.3.1
imageio                       2.36.0
imagesize                     1.4.1
importlib_metadata            7.1.0
importlib_resources           6.4.0
iniconfig                     2.0.0
ipdb                          0.13.13
ipykernel                     6.29.3
ipython                       8.22.2
jax                           0.4.25
jaxlib                        0.4.25
jedi                          0.19.1
Jinja2                        3.1.3
joblib                        1.3.2
jsonpatch                     1.32
jsonpointer                   2.1
jupyter_client                8.6.1
jupyter_core                  5.7.2
keras                         3.8.0
kiwisolver                    1.4.5
libclang                      18.1.1
libmambapy                    1.5.3
llvmlite                      0.44.0
Markdown                      3.6
markdown-it-py                3.0.0
MarkupSafe                    2.1.5
marshmallow                   3.23.1
matplotlib                    3.9.0
matplotlib-inline             0.1.6
mdurl                         0.1.2
menuinst                      2.0.2
ml-dtypes                     0.4.1
mpmath                        1.3.0
msgpack                       1.0.8
mujoco                        3.2.4
mujoco-py                     2.1.2.14
multidict                     6.0.5
multiprocess                  0.70.16
namex                         0.0.8
nest_asyncio                  1.6.0
networkx                      3.2.1
numba                         0.61.0
numpy                         1.26.4
numpydoc                      1.8.0
nvidia-cublas-cu12            12.1.3.1
nvidia-cuda-cupti-cu12        12.1.105
nvidia-cuda-nvrtc-cu12        12.1.105
nvidia-cuda-runtime-cu12      12.1.105
nvidia-cudnn-cu12             8.9.2.26
nvidia-cufft-cu12             11.0.2.54
nvidia-curand-cu12            10.3.2.106
nvidia-cusolver-cu12          11.4.5.107
nvidia-cusparse-cu12          12.1.0.106
nvidia-nccl-cu12              2.20.5
nvidia-nvjitlink-cu12         12.4.99
nvidia-nvtx-cu12              12.1.105
opencv-python                 4.9.0.80
openpyxl                      3.1.5
opt-einsum                    3.3.0
optax                         0.2.2
optree                        0.14.0
orbax-checkpoint              0.5.7
packaging                     23.1
pandas                        2.2.1
parso                         0.8.4
pastel                        0.2.1
peft                          0.17.0
pexpect                       4.9.0
pickleshare                   0.7.5
pillow                        10.3.0
pip                           23.3.1
platformdirs                  3.10.0
plotly                        5.22.0
pluggy                        1.5.0
powerlaw                      1.5
prompt-toolkit                3.0.42
protobuf                      4.25.3
psutil                        5.9.8
ptyprocess                    0.7.0
pure-eval                     0.2.2
pyarrow                       20.0.0
pycosat                       0.6.6
pycparser                     2.21
pygame                        2.6.0
pyglet                        2.0.18
Pygments                      2.17.2
pylev                         1.4.0
PyOpenGL                      3.1.7
pyparsing                     3.1.2
pysimdjson                    6.0.2
PySocks                       1.7.1
pystan                        3.10.0
pytest                        8.3.3
pytest-instafail              0.3.0
python-dateutil               2.9.0
pytz                          2024.1
PyWavelets                    1.7.0
PyYAML                        6.0.1
pyzmq                         25.1.2
regex                         2024.11.6
requests                      2.32.4
rich                          13.7.1
rlax                          0.1.6
ruamel.yaml                   0.17.21
safetensors                   0.5.2
scikit-learn                  1.4.1.post1
scipy                         1.12.0
seaborn                       0.13.2
sentry-sdk                    1.44.0
setproctitle                  1.3.3
setuptools                    68.2.2
six                           1.16.0
smmap                         5.0.1
snowballstemmer               2.2.0
Sphinx                        8.1.3
sphinx-rtd-theme              3.0.2
sphinxcontrib-applehelp       2.0.0
sphinxcontrib-devhelp         2.0.0
sphinxcontrib-htmlhelp        2.1.0
sphinxcontrib-jquery          4.1
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          2.0.0
sphinxcontrib-serializinghtml 2.0.0
stack-data                    0.6.2
swig                          4.2.1
sympy                         1.12
tabulate                      0.9.0
tenacity                      8.5.0
tensorboard                   2.18.0
tensorboard-data-server       0.7.2
tensorflow                    2.18.0
tensorflow-probability        0.25.0
tensorstore                   0.1.56
termcolor                     2.5.0
threadpoolctl                 3.3.0
tokenizers                    0.21.1
toolz                         0.12.1
torch                         2.3.0
torch_geometric               2.5.1
torchsummary                  1.5.1
torchvision                   0.18.0
tornado                       6.4
tqdm                          4.67.1
traitlets                     5.14.2
transformers                  4.52.3
truststore                    0.8.0
typing_extensions             4.11.0
tzdata                        2024.1
urllib3                       2.1.0
wandb                         0.16.5
wcwidth                       0.2.13
webargs                       8.6.0
Werkzeug                      3.0.1
wheel                         0.41.2
wrapt                         1.17.2
xxhash                        3.5.0
yarl                          1.9.4
zipp                          3.18.1
zstandard                     0.19.0
